component/component.go

```diff
@@ -125,3 +125,10 @@ type HTTPComponent interface {
 	// will receive a request to just `/metrics`.
 	Handler() http.Handler
 }
+
+// ClusterComponent is an extension interface for components which implement
+// new behavior whenever the Flow controller is running in clustered mode.
+type ClusterComponent interface {
+	Component
+	ClusterWatch() bool
+}
```

Comment:
I'm not sure about this long term, but I'm OK with it for the scope of the initial PR.

I'm still interested some kind of signaling mechanism so components can register/unregister themselves for stuff like cluster events when their Update method is called. 

This could be handled by passing a context.Context with some special values in it, or maybe extra arguments to Update to allow us to give that kind of data to components.

Reply:
I've removed this for now; components call directly the Register method through their Options. I'm not sure if this the way we'll go for, but it's much better than this ugly solution.